File: app/src/main/java/com/scatl/uestcbbs/module/darkroom/adapter/DarkRoomAdapter.java
package com.scatl.uestcbbs.module.darkroom.adapter;

import android.graphics.Color;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.DarkRoomBean;

/**
 * author: sca_tl
 * date: 2021/4/10 12:56
 * description:
 */
public class DarkRoomAdapter extends BaseQuickAdapter<DarkRoomBean, BaseViewHolder> {
    public DarkRoomAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, DarkRoomBean item) {
        helper.setText(R.id.item_dark_room_user_name, item.username)
                .setText(R.id.item_dark_room_action, item.action)
                .setText(R.id.item_dark_room_action_time, item.actionTime)
                .setText(R.id.item_dark_room_date_line, item.dateline)
                .setText(R.id.item_dark_room_reason, item.reason)
                .addOnClickListener(R.id.item_dark_room_user_name);

        TextView action = helper.getView(R.id.item_dark_room_action);
        if ("禁止发言".equals(item.action)) {
            action.setTextColor(Color.parseColor("#CCAF12"));
        } else if("禁止访问".equals(item.action)) {
            action.setTextColor(Color.parseColor("#CC4347"));
        }

    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/darkroom/adapter/DarkRoomAdapter.kt
package com.scatl.uestcbbs.module.darkroom.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemDarkRoomBinding
import com.scatl.uestcbbs.entity.DarkRoomBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder

/**
 * Created by sca_tl at 2023/6/6 16:06
 */
class DarkRoomAdapter : PreloadAdapter<DarkRoomBean, ItemDarkRoomBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemDarkRoomBinding {
        return ItemDarkRoomBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemDarkRoomBinding>, position: Int, item: DarkRoomBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.userName.text = item.username
        holder.binding.actionTime.text = item.actionTime
        holder.binding.dateLine.text = item.dateline
        holder.binding.reason.text = item.username

        holder.binding.action.apply {
            text = item.action
            setTextColor(
                when(text) {
                    "禁止发言" -> { Color.parseColor("#CCAF12") }
                    "禁止访问" -> { Color.parseColor("#CC4347") }
                    else -> { Color.parseColor("#DDDDDD") }
                }
            )
        }
    }
}